### PR TITLE
lightdm: enable the accounts daemon to find dbus interface

### LINKED ISSUE #45059: LightDM: No such interface 'org.freedesktop.DisplayManager.AccountsService'
## Issue description

LightDM seems not installing dbus interface of AccountsService.

Systemd journal:
```
lightdm[1335]: Error updating user /org/freedesktop/Accounts/User1000: GDBus.Error:org.freedesktop.DBus.Error.InvalidArgs: No such interface 'org.freedesktop.DisplayManager.AccountsService'
```

In `lightdm` package, it is contained:
```
% nix build -f '<nixpkgs>' lightdm
% ls result/share/dbus-1/interfaces | grep -i accounts
org.freedesktop.DisplayManager.AccountsService.xml
```
But in the current system:
```
% ls /run/current-system/sw/share/dbus-1/interfaces | grep -i accounts
org.freedesktop.Accounts.User.xml@
org.freedesktop.Accounts.xml@
```

### Steps to reproduce

```
services.xserver.displayManager.lightdm.enable = true;
```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.18.0, NixOS, 18.09.git.8483983 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixos-18.09pre148970.b5b11e9da43"`
 - channels(mnacamura): `"nixpkgs-18.09pre148461.bf1b50cbc8f"`
 - nixpkgs: `/home/mnacamura/repos/nixpkgs`

### PR DESCRIPTION
**Pull request by mnacamura**
###### Motivation for this change

Fix #45059

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/x11/display-managers/lightdm.nix [hunk 1]
```diff
@@ -212,6 +212,10 @@ in
     # lightdm uses the accounts daemon to rember language/window-manager per user
     services.accounts-daemon.enable = true;
 
+    # Enable the accounts daemon to find lightdm's dbus interface
+    environment.pathsToLink = [ "/share/accountsservice" ];
```
Comment (jtojnar) [inline nixos/modules/services/x11/display-managers/lightdm.nix]: I think this should go to `accountsservice.nix`. There might be other extensions than lightdm.

File: nixos/modules/services/desktops/accountsservice.nix [hunk 2]
```diff
@@ -36,11 +36,14 @@ with lib;
...
     systemd.packages = [ pkgs.accountsservice ];
 
-    systemd.services.accounts-daemon= {
+    systemd.services.accounts-daemon = {
 
       wantedBy = [ "graphical.target" ];
 
-    } // (mkIf (!config.users.mutableUsers) {
+      # Accounts daemon looks for dbus interfaces in $XDG_DATA_DIRS/accountsservice
+      environment.XDG_DATA_DIRS = "/run/current-system/sw/share";
```
Comment (jtojnar) [inline nixos/modules/services/desktops/accountsservice.nix]: I think something like `${config.system.path}/share` should be better.

File: nixos/modules/services/x11/display-managers/lightdm.nix (hunk 1 above)
Comment (mnacamura) [reply nixos/modules/services/x11/display-managers/lightdm.nix]: Right. Done.

File: nixos/modules/services/desktops/accountsservice.nix (hunk 2 above)
Comment (mnacamura) [reply nixos/modules/services/desktops/accountsservice.nix]: Done. Thank you for your review!

Comment (michaelpj) [reply nixos/modules/services/desktops/accountsservice.nix]: Hm, shouldn't systemd system services run in an environment that has `XDG_DATA_DIRS` set appropriately for the current system? I'm surprised more things don't break if that not true.

Comment (jtojnar) [reply nixos/modules/services/desktops/accountsservice.nix]: The variables are often set by login shell, so the services cannot see them. It is probably good thing, as the unclean environment could produce unpredictable behaviour.

**Comment by jtojnar** [comment]
Looks great, thanks.

**Pull request merged**